### PR TITLE
Migration distro and image from zeus to hardknott

### PR DESCRIPTION
**Pull request by motto-phytec**
an update from zeu to hardknott is not possible directly, because the following has changed
1. RAUC Bundle signing type has changed from plain (zeus) to the new type variety (hardknott)
2. RAUC certificate structure from selfsigned rauc ca to phytec main ca with rauc intermediate
3. The shadow_command has changed from hardcoded in awsclient version 1.3 (zeus) to configurable in aws/config.json by the awsclient 1.5 (hardknott)

### CODE REVIEW

File: conf/distro/ampliphy-vendor-connagtive-migration.conf [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+DISTRO = "ampliphy-vendor-connagtive-migration"
+DISTRO_NAME = "ampliphy Vendor Connagtive Migration from zeus to hardknott"
+
+PREFERRED_VERSION_phytec-dev-ca = "1.2"
+
+# RAUC Bundle Format from zeus
+RAUC_BUNDLE_FORMAT = "plain"
+RAUC_KEY_FILE ?= "${CERT_PATH}/rauc/private/development-1.key.pem"
+RAUC_CERT_FILE ?= "${CERT_PATH}/rauc/development-1.cert.pem"
```
Comment (mschwan-phytec) [inline conf/distro/ampliphy-vendor-connagtive-migration.conf]: These RAUC variables should be set in the migration bundle recipe, not the distro.

File: conf/distro/ampliphy-vendor-connagtive.conf [hunk 2]
```diff
@@ -1,13 +1,7 @@
...
 
-#add protection shield level
-#shieldlow - shieldmedium - shieldhigh
-DISTRO_FEATURES += "protectionshield"
-PROTECTION_SHIELD_LEVEL = "shieldmedium"
-OVERRIDES_append = ":protectionshield:${PROTECTION_SHIELD_LEVEL}"
-
-#password or authenticator for console and sshd
-CONNAGTIVE_ROOT_AUTHENTICATION = "password"
+PREFERRED_VERSION_phytec-dev-ca = "1.2"
```
Comment (mschwan-phytec) [inline conf/distro/ampliphy-vendor-connagtive.conf]: Setting the preferred version of phytec-dev-ca to 1.2 in the normal connagtive distro seems wrong. We only need it in the migration distro, right?

File: conf/distro/common-secure-connagtive.inc [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
```
Comment (mschwan-phytec) [inline conf/distro/common-secure-connagtive.inc]: Remove this empty line.

**Review by mschwan-phytec — CHANGES_REQUESTED** [review]

File: conf/distro/ampliphy-vendor-connagtive.conf (hunk 2 above)
Comment (motto-phytec) [reply conf/distro/ampliphy-vendor-connagtive.conf]: the main certificate and rauc intermediate certificate are equal in version 1.2 and 1.3. The phytec-dev-ca is inthe ../build folder. So different builds with the same source use the same phytec-dev-ca folder.  This is the main reason for the same phytec-dev-ca version in all distros.

File: conf/distro/ampliphy-vendor-connagtive-migration.conf (hunk 1 above)
Comment (motto-phytec) [reply conf/distro/ampliphy-vendor-connagtive-migration.conf]: is moved to the migration bundle and migration distro is removed

File: conf/distro/common-secure-connagtive.inc (hunk 3 above)
Comment (motto-phytec) [reply conf/distro/common-secure-connagtive.inc]: is removed

File: conf/distro/ampliphy-vendor-connagtive.conf (hunk 2 above)
Comment (mschwan-phytec) [reply conf/distro/ampliphy-vendor-connagtive.conf]: But then, for all connagtive related distros to use the same phytec-dev-ca version in this layer, shouldn't we move that setting to the `common-secure-connagtive.inc`?

**Review by mschwan-phytec — CHANGES_REQUESTED** [review]
What do you think about the bundle and image names? Maybe we should add "zeus-to-hardknott" to their names to make the affected versions clear. Although that makes them look quite bulky, it assures we don't run into name clashes for any possible future migrations.

File: conf/distro/ampliphy-vendor-connagtive.conf (hunk 2 above)
Comment (motto-phytec) [reply conf/distro/ampliphy-vendor-connagtive.conf]: yes this is right. I move this to the common-secure-connagtive.inc.

**Comment by motto-phytec** [comment]
I change the names for the migration image and bundle and add zeus-to-hardknott in the receipes names.

**Review by mschwan-phytec — APPROVED** [review]

**Pull request merged**